### ISSUE TITLE
Rebuild vocabulary browser with jskos-vue

### ISSUE THREAD

**Issue by nichtich** [gbv/bartoc.org#129]
With release 0.2.0 of [jskos-vue](https://github.com/gbv/jskos-vue) the vocabulary browser can be rewritten by using this library.

**Comment by rodolv-commons** [comment]
The vocabulary browser now has a clearer layout. 

- Search results appear only after typing, while top concepts remain in the concept tree. When a concept is selected, the tree stays visible on the left and the concept details appear on the right.
- Users can select concepts from search results, the tree, or the hierarchy shown in the details panel. The selected concept is also stored in the page URL, so it can be shared directly.
- If a terminology provides multiple APIs, a Data source selector is shown. This is useful because different APIs may provide different labels, details, or hierarchy data.

Examples:
- [Browse the DDC vocabulary](https://dev.bartoc.org/en/node/241#content)
- [Open DDC class 20 directly](https://dev.bartoc.org/en/node/241?uri=http%3A%2F%2Fdewey.info%2Fclass%2F20%2Fe23%2F#content)

Ready for review

**Comment by mroehl86** [comment]
I have a few additional comments and suggestions:

1. The hierarchy is already shown in the concept tree on the left. Do we also need to show the hierarchy in the concept details? If we keep it, maybe we could add headings like “Broader concepts” and “Narrower concepts”. At the moment, it is not always clear which information belongs to the selected concept. For example, the identifier can look like it belongs to the broader concept above it.

<img width="945" height="476" alt="Image" src="https://github.com/user-attachments/assets/f628010f-d8e8-4d2b-89bf-a5e2c07b2956" />


 For example, the concept details could be structured like this:

**Concept details**

40 Language

Identifier
http://dewey.info/class/40/e23/

Broader concept
4 Language

Narrower concepts
400 Language
401 Philosophy & theory
402 Miscellany
....

2. For search results with identical or very similar labels, it might be helpful to also show the broader concept. This could make results such as “404 Special topics” and “604 Special topics” easier to distinguish.

<br>

<img width="945" height="359" alt="Image" src="https://github.com/user-attachments/assets/9a936e9e-6e26-4469-8d45-058f0250d1f8" />

<br><br>

3. The bullet points in the concept details make the metadata a bit hard to read. Maybe the metadata fields could be shown without bullet points, and bullets could only be used for multiple values, such as “Alternative Labels”.

<br>

<img width="945" height="436" alt="Image" src="https://github.com/user-attachments/assets/c7e00519-7b78-49e9-88ea-a77254e34e30" />

<br><br>

4. The Search and Data source fields could be better aligned. They have slightly different heights, and the Data source field seems a bit too wide.

<br>
 
<img width="945" height="258" alt="Image" src="https://github.com/user-attachments/assets/d7570c53-c54e-4ab5-afbd-2de43c46d814" />

<br><br>

5. Maybe the “Browse concepts” and “Concept details” section could have a more visible box. The current thin border lines do not separate the section very clearly from the rest of the page.

**Comment by rodolv-commons** [comment]
thanks @mroehl86 , I will go through all the points you mentioned.

**Comment by rodolv-commons** [comment]
Thanks, these are helpful suggestions!
1. I agree that this is partly redundant because the hierarchy is already available in the tree on the left. I kept it because it was present in the previous implementation and is also used by uri.gbv.de/terminology. It may help users understand the current context and navigate related concepts directly from the details. Is there another reason why it was originally designed this way? For now, I would keep it but add the clearer Broader concept and Narrower concepts sections you suggested.
2. I think this belongs in jskos-vue, because it controls how the search results are rendered. It may also require additional hierarchy data and a clear design for longer result lists. In the DDC example, the notation prefixes 404 and 604 already help distinguish the two results. I would therefore keep this out of scope for this issue, but I am open to addressing it in jskos-vue once we have a concrete UX proposal.
3. I agree. Single metadata values should not use bullet points. Bullets are more useful for fields with multiple values, such as Alternative Labels.
4. Agreed. I will align the Search and Data source fields and make the Data source field narrower.
5. Agreed. I will make the Browse concepts and Concept details boxes easier to distinguish.

**Comment by nichtich** [comment]
> 1. The hierarchy is already shown in the concept tree on the left. Do we also need to show the hierarchy in the concept details? 

No, but it depends on the terminology type. This needs to be tested with different types of terminologies (classification, thesaurus, name authority file) and sizes .
  
> 2. For search results with identical or very similar labels, it might be helpful to also show the broader concept.

I'd keep it this way. This minor annoyance is only relevant to large classifications such as DDC. 

----

More comments: 

- Wikidata does not show browser: https://dev.bartoc.org/en/node/1940#content
- Search and switching data source seems to not work at <https://dev.bartoc.org/en/node/430#content>
- if there is only one service (data source) it should still be shown as plain text to inform, where the data comes from. Better always show "Search" and "Data Source". If no search is possible note "Terminology search not supported" instead of the search input field.
- Show a loading indicator. Some data sources are slow or not reachable at all, so we need timeout, error message and "retry" button if service is not reachable.
- cocoda-sdk does not support all API types. Unsupported types should be disabled in data source dropdown. The component can have a whitelist of supported API types.

**Comment by rodolv-commons** [comment]
Two notes from testing the data sources:

- Wikidata: The JSKOS service provides suggest, concepts, and data, but no top, narrower, or ancestors endpoint. The previous BARTOC browser required top concepts, so Wikidata was not shown at all. BARTOC now accepts search-only services and can show search and concept details without a tree.
  If a Wikidata hierarchy or similar is wanted, this should probably be discussed in [wikidata-jskos](https://github.com/gbv/wikidata-jskos). It would require a clear definition of top concepts and support for hierarchy requests. Performance and HTTP 429 responses (found many times during testing) should also be investigated there.

- GND/DANTE: DANTE is a supported JSKOS API, but its /voc endpoint does not return GND for BARTOC 430 or its alternative identifiers. This is not an unsupported API type: the service simply does not contain this terminology. I think the DANTE endpoint should therefore be removed from record 430. It does make any sense? 

Lobid remains the working source for GND. Other API types not supported by cocoda-sdk are shown but disabled in the data source dropdown.

If you notice any other edge cases or limitations, please report them here.
